homeassistant/components/overkiz/__init__.py

```diff
@@ -96,6 +105,11 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
             device,
         )
 
+        if platform := OVERKIZ_DEVICE_TO_PLATFORM.get(
+            device.widget
+        ) or OVERKIZ_DEVICE_TO_PLATFORM.get(device.ui_class):
```

Comment:
Should this be an `and`?

Reply:
We use this logic to match the widget or ui_class of a Overkiz device. Normally we match on the ui_class, however in some cases we need to override this and match on the widget.

The `or` here is a conscious choice.

Reply:
How is that going to work if `platform` is None?

Reply:
If `platform` is None (because it cannot find a match in `OVERKIZ_DEVICE_TO_PLATFORM`), it won't add the specific device to the platform list.

If there is a match, it will add the device to the list for the specific platform (`platforms[platform].append(device)`)  and thus it will be loaded in in `async_setup_entry` for the specific entity.
Or didn't I understood your question correctly?

Reply:
Nevermind, I'm reading it wrong since its split between two lines 🤣

Reply:
I was reading it as `if (platform := OVERKIZ_DEVICE_TO_PLATFORM.get(device.widget) or OVERKIZ_DEVICE_TO_PLATFORM.get(device.ui_class):`

Reply:
tbh I wasn't sure if I did prefer the new notation for this long statement